### DISCUSSION TITLE
Any way to remove auto-pair brackets?

### DISCUSSION THREAD

**Discussion by etiennelacoursiere** [zed-industries/zed#7612] [Feature Requests]
Hi! :) 

I looked around in the docs but didn't find an answer. I know i might be a minority but i really dislike auto-pair bracket. Meaning if i type `{` the editor automatically adds a pairing `}` same thing for quotes. 

Is there any config do disabled it? 

Thank you!

**Comment by SomeoneToIgnore** [comment]
There are two things that could force autocloses, here are the relevant default settings.json entries:

* Zed functionality, defined for every language like https://github.com/zed-industries/zed/blob/f60bebb93ba9cce6e17949450cdba56a1e14dc7c/crates/zed/src/languages/rust/config.toml#L6-L13
```
// Whether to automatically type closing characters for you. For example,
// when you type (, Zed will automatically add a closing ) at the correct position.
"use_autoclose": true,
```

* LSP one, that could add brackets or not, depending on the underlying implementation. It might also format the code, or do something else after typing:
```
// Whether to use additional LSP queries to format (and amend) the code after
// every "trigger" symbol input, defined by LSP server capabilities.
"use_on_type_format": true,
```

I would really appreciate if you could add that into the docs: https://github.com/zed-industries/zed/tree/main/docs/src where you see it should fit.

**Reply by mbylstra** [reply]
Could you please give some more concrete instructions on how to turn off brackets autocomplete? I don't understand these instructions. Could you give an actual example of something I would be in my settings.json? I never want brackets autocomplete, for any language. In vscode it's just a simple setting `"editor.autoClosingBrackets": "never"`. Is there something similar?

**Reply by etiennelacoursiere** [reply]
Pretty sure `"use_autoclose": false` is what you want

**Reply by mbylstra** [reply]
Ah, cheers, I got confused by the rust snippet, thinking I needed to define it for every language.
